### DISCUSSION TITLE
Pending rke2-coredns-rke2-coredns pods due to required antiaffinity rule

### DISCUSSION THREAD

**Discussion by MrBlaise** [rancher/rke2#7262] [Q&A] [answered]
**Environmental Info:**
RKE2 Version: `v1.31.1+rke2r1`

Node(s) CPU architecture, OS, and Version: 

`6.8.0-48-generic #48~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Mon Oct  7 11:24:13 UTC 2 x86_64 x86_64 x86_64 GNU/Linux`

Cluster Configuration:

3 server, 3 agents

**Describe the bug:**

I have 6 nodes, rke2-coredns tries to deploy 9 instances, but the antiaffinity rule does not allow deploying 2 coredns pods next to each other, due to this: https://github.com/rancher/rke2-charts/blob/main-source/packages/rke2-coredns/generated-changes/patch/values.yaml.patch#L56-L65

The reason it tries to deploy 9 instances is because rke2-coredns-autoscaler is telling it to do so, because I have a lot of CPU cores on the machines.

Why is this antiaffinity rule required and not preferred?

**Comment by brandond** [comment, marked as answer]
You could probably better tune the autoscaler for your use case, or disable it entirely if you prefer. 9 pods for 6 nodes is probably overkill.

The anti-affinity rule exists because HA is not improved by having a bunch more pods all running on the same node.
